File: SoftUni Lessons/Python Development/Python Fundamentals September 2019/Problems And Files/13 EXERCISE FUNCTIONS - Дата 11-ти октомври, 1430 - 1730/04. Odd and Even Sum.py
"""
Functions - Exercise
Check your code: https://judge.softuni.bg/Contests/Practice/Index/1728#3

SUPyF2 Functions-Exercise - 04. Odd and Even Sum

Problem:
You will receive a single number.
You have to write a function that returns the sum of all even and all odd digits from that number
 as a single string like in the examples below.

Examples:
Input:	                Output:
1000435	                Odd sum = 9, Even sum = 4
3495892137259234	    Odd sum = 54, Even sum = 22
"""


def even_and_odd_sum(number):
    odd_digits_list = [int(digit) for digit in number if digit % 2 != 0]
    even_digits_list = [int(digit) for digit in number if digit % 2 == 0]
    return f"Odd sum = {sum(odd_digits_list)}, Even sum = {sum(even_digits_list)}"


print(even_and_odd_sum([int(digit) for digit in input()]))

File: SoftUni Lessons/JavaScript Development/JS Fundamentals January 2020/15. EXERCISE FUNCTIONS/04. Odd And Even Sum.js
function oddAndEvenSum(input) {
    // Mask
    const getSum = (array) => {
        let even = 0; let odd = 0;
        for (let number of array) { number % 2 === 0 ? even += number : odd += number; }
        return [even, odd]
    };

    let sums = getSum(String(input).split('').map(Number));
    console.log(`Odd sum = ${sums[1]}, Even sum = ${sums[0]}`);
}

oddAndEvenSum(1000435);
// Should return: Odd sum = 9, Even sum = 4

oddAndEvenSum(3495892137259234);
// Should return: Odd sum = 54, Even sum = 22